app/models/business_unit.rb

```diff
@@ -39,11 +40,17 @@ class BusinessUnit < Team
   has_many :managers, through: :manager_user_roles, source: :user
   has_many :responders, through: :responder_user_roles, source: :user
   has_many :approvers, through: :approver_user_roles, source: :user
+  has_many :category_roles, foreign_key: :team_id, class_name: 'TeamCategoryRole'
 
   scope :managing, -> { where(role: 'manager') }
   scope :approving, -> { where(role: 'approver') }
   scope :responding, -> { where(role: 'responder') }
 
+
+  def self.responding_for_category(category)
+    joins(:category_roles).where('team_category_roles.category_id = ? and team_category_roles.respond = ?', category.id, true)
+  end
```

Comment:
Looking at this, and at the table definition below, I was imagining that this would be implemented differently. Instead of, for example, an entry for BMT which would look like:

id | category_id | team_id | view? | edit? | manage? | respond? | approve?
-|-|-|-|-|-|-|-
1 | 1 | 1 | true | true | true | false | false

I thought we might have:

id | category_id | team_id | role
-|-|-|-
1 | 1 | 1 | 'viewing'
1 | 1 | 1 | 'editing'
1 | 1 | 1 | 'managing'

Some of the benefits:
- adding roles without needing a migration
- if we use AR enums for the role column, we get free scopes like `managing` or `responding` on `TeamCategoryRole` which would simplify the above method

Not sure if this design speaks to you or not.